drivers/dht/dht.c

```diff
@@ -43,13 +43,32 @@
 /* If an expected pulse is not detected within 1000µs, something is wrong */
 #define TIMEOUT                     (1000U)
 /* The DHT sensor cannot measure more than once a second */
-#define DATA_HOLD_TIME              (1000U * US_PER_MS)
+#define DATA_HOLD_TIME              (US_PER_SEC)
 
 static inline void _reset(dht_t *dev)
 {
     gpio_init(dev->params.pin, GPIO_OUT);
     gpio_set(dev->params.pin);
+}
 
+/**
+ * @brief   Wait until the pin @p pin has level @p expect
+ *
+ * @param   pin     GPIO pin to wait for
+ * @param   expect  Wait until @p pin has this logic level
+ * @param   timeout Timeout in µs
+ *
+ * @retval  0       Success
+ * @retval  -1      Timeout occurred before level was reached
+ */
+static inline int _wait_for_level(gpio_t pin, int expect, unsigned timeout)
+{
+    while ((gpio_read(pin) != expect) && timeout) {
```

Comment:
not sure this works, bc `gpio_read` may return anything `> 0` for HIGH, i.e., must not be `1` in all cases. That's why I had this rather ugly line `((gpio_read(pin) > 0) == set)` in my suggested code, that should work for cases there `gpio_read()` returns `42` (for example) for HIGH.